gzip.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package gzip
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/flamego/flamego"
+	"github.com/klauspost/compress/gzip"
```

Comment:
We should be using stdlib's gzip package, the optimization of done in "github.com/klauspost/compress/gzip" has been merged to stdlib in recent Go versions IIRC.